### PR TITLE
single parameter fit

### PR DESCRIPTION
**Pull request by tgiani**
implementation of single parameter fits

### CODE REVIEW

File: src/smefit/postfit/__init__.py [hunk 1]
```diff
@@ -35,11 +36,16 @@ def from_file(cls, runcard_folder, run_card_name):
         result_ID = config.get("result_ID", run_card_name)
 
         # get the chi2_threshold, by default is 3.0
-        if "chi2_threshold" not in config:
+        if "chi2_threshold" not in config and not config["single_parameter_fits"]:
```
Comment (giacomomagni) [inline src/smefit/postfit/__init__.py:39]: mmm why did you change this check?

Comment (giacomomagni) [inline src/smefit/postfit/__init__.py:39]:
```suggestion
        if "chi2_threshold" not in config and not self.single_parameter_fits:
```

File: src/smefit/optimize/ns.py [hunk 2]
```diff
@@ -289,5 +297,14 @@ def save(self, result):
             for c in self.coefficients.name:
                 values[c].append(self.coefficients[c].value)
 
-        with open(self.results_path / "posterior.json", "w", encoding="utf-8") as f:
+        posterior_file = self.results_path / "posterior.json"
+
+        # if it s a single parameter fit check if the posterior file is already present and in case it is update it
```
Comment (giacomomagni) [inline src/smefit/optimize/ns.py:302]: this code is a bit repeated can you move it to the same place both for NS and MC?

File: src/smefit/optimize/ns.py [hunk 3]
```diff
@@ -105,6 +107,11 @@ def from_dict(cls, config):
             )
         coefficients = CoefficientManager.from_dict(config["coefficients"])
 
+        if "single_parameter_fits" not in config:
```
Comment (giacomomagni) [inline src/smefit/optimize/ns.py:110]: here and in MC you can use:

```suggestion
       single_parameter_fits = config.get("single_parameter_fits", False)
```

File: src/smefit/runner.py [hunk 4]
```diff
@@ -104,21 +104,57 @@ def ns(self):
...
         """
         Run a fit with MC
         """
         config = self.run_card
-        opt = MCOptimizer.from_dict(config)
-        opt.run_sampling()
-        opt.save()
+
+        # single parameter fits
+        if config["single_parameter_fits"]:
```
Comment (giacomomagni) [inline src/smefit/runner.py:145]: also here maybe you can avoid repetition...

Comment (tgiani) [reply src/smefit/runner.py:145]: mm you r right maybe then I should code another function just for the single parameter fits case

**Review by giacomomagni — COMMENTED** [review]
Okay looks good. Before merging we need to: 
* Merge `MC_fit` here to avoid conflicts.
* Add some simple docstring to the new functions, at some point we will have to do it ... 
* Run black to keep the correct formatting
* Update the Readme with the correct instructions.
(not really needed here, but to be done before Wednesday... )
